src/containers/Tenant/Schema/SchemaTree/SchemaTree.tsx

```diff
@@ -50,33 +58,66 @@ export function SchemaTree(props: SchemaTreeProps) {
         return childItems;
     };
 
+    React.useEffect(() => {
+        if (isSuccess) {
+            setNavigationTreeKey(requestId);
+        }
+    }, [isSuccess, requestId]);
+
     React.useEffect(() => {
         // if the cached path is not in the current tree, show root
         if (!currentPath?.startsWith(rootPath)) {
             onActivePathUpdate(rootPath);
         }
     }, [currentPath, onActivePathUpdate, rootPath]);
 
+    const handleCreateDirectoryClose = () => {
+        setCreateDirectoryOpen(false);
+    };
+
+    const handleCreateDirectorySubmit = (child: string) => {
+        createDirectory({database: parent, path: `${parent}/${child}`});
+
+        onActivePathUpdate(`${parent}/${child}`);
+
+        setCreateDirectoryOpen(false);
```

Comment:
same here. I would set some `loading` state to dialog `Apply` buttons and close dialog on request success. If request fails we may show an error inside the dialog's body.